README.md

```diff
@@ -1 +1,152 @@
+commands, providing sensible / flexible defaults, and support various scenarios
+("archetypes") for your common use cases across multiple projects.
+
+## Tasks
+
+Builder can run 1+ tasks based out of `package.json` `scripts`. For a basic
+scenario like:
+
+```js
+{
+  "scripts": {
+    "foo": "echo FOO",
+    "bar": "echo BAR"
+  }
+}
+```
+
+Builder can run these tasks individually:
+
+```sh
+$ builder run foo
+$ builder run bar
+```
+
+Sequentially via `||` or `&&` shell helpers:
+
+```sh
+$ builder run foo && builder run bar
+```
+
+Concurrently via the Builder built-in `concurrent` command:
+
+```sh
+$ builder concurrent foo bar
+```
+
+With `concurrent`, all tasks continue running until they all complete _or_
+any task exits with a non-zero exit code, in which case all still alive tasks
+are killed and the Builder process exits with the error code.
+
+## Archetypes
+
+Archetypes deal with common scenarios for your projects. Like:
+
+* [builder-react-component][]: A React component
+* A React application server
+* A Chai / jQuery / VanillaJS widget
+
+Archetypes typically provide:
+
+* A `package.json` with `builder`-friendly `script` tasks.
+* Dependencies and dev dependencies to build, test, etc.
+* Configuration files for all `script` tasks.
+
+In most cases, you won't need to override anything. But, if you do, pick the
+most granular `scripts` command in the archetype you need to override and
+define _just that_ in your project's `package.json` `script` section. Copy
+any configuration files that you need to tweak and re-define the command.
+
+### Special Archetype Tasks
+
+Archetypes use conventional `scripts` task names, except for the following
+special cases:
+
+* `"npm:postinstall"
+* `"npm:preversion"
+* `"npm:version"
+* `"npm:test"`
+
+These tasks are specifically actionable during the `npm` lifecycle, and
+consequently, the archetype mostly ignores those for installation by default,
+offering them up for actual use in _your_ project.
+
+As an **additional restriction**, non-`npm:FOO`-prefixed tasks with the same
+name (e.g., `FOO`) _may_ call then `npm:`-prefixed task, but _not_ the other
+way around. So
+
+```js
+// Good / OK
+"npm:test": "builder run test-frontend",
+"test": "builder run npm:test",
+
+// Bad
+"npm:test": "builder run test",
+"test": "builder run test-frontend",
+````
+
+## Package Script Commands
+
+The underyling concept here is that `builder` `script` commands simple _are_
+NPM-friendly `package.json` `script` commands. Pretty much anything that you
+can execute with `npm run FOO` can be executed with `builder run FOO`.
+
+### Tips & Tricks
+
+#### Terminal Color
+
+Builder uses `exec` under the hood with piped `stdout` and `stderr`. Programs
+typically interpret the piped environment as "doesn't support color" and
+disable color. Consequently, you typically need to set a "**force color**"
+option on your executables in `scripts` commands if they exist.
```

Comment:
Your intuition is correct -- `spawn` does very poorly with arbitrary shell commands as you have to split the script + args yourself, and this parsing is hard to get right by yourself. Some background

* Some utils like `concurrently` just do a `ARG_STRING.split(" ")` and pray for the best.
* There's even attempts at a tool to do this for you that still falls short of the mark: https://github.com/binocarlos/spawn-args

So... `exec` just works. And doing the colors thing really hasn't been hard at all so far for any of the commands in our react component archetype.